golem-common/src/model/mod.rs

```diff
@@ -858,14 +861,20 @@ impl IntoValue for WorkerStatus {
 
 #[derive(Clone, Debug, PartialEq, Encode, Decode)]
 pub enum WorkerInvocation {
-    ExportedFunction {
+    ExportedFunctionV1 {
```

Comment:
If this is done for oplog serialization purposes, I would propose to have a separate data type for that with mapping between the two.
If possible, I think it would make our lives much easier to not have multiple versions in domain types.

Reply:
With bincode we must introduce a new constructor to keep compatibility. 
I would like to eventually switch to my binary serializer for these internal formats which is focusing on supporting more data evolution cases but haven't had time to fully finish and test it yet.

Reply:
my point is that we are using this type in other places / our domain.
I would propose having the extra constructors for bincode on a seperate type that is getting converted from/to our domain type when we need to serialize it.
This way the extra cases will not be visible to other code working with it (as they should not be using them directly if I'm understanding this correctly)